### ISSUE TITLE
convert foo.odt foo.pdf fails (delegates do not support shell commands)

### ISSUE THREAD

**Issue by Lekensteyn** [ImageMagick/ImageMagick#3]
The ImageMagick-6 branch has "support" for conversion between file formats (trunk / master and the delegates.xml file at https://www.imagemagick.org/script/resources.php is older). This unfortunately does not work as expected.

By invoking `convert foo.odt foo.pdf`, a PDF is generated in `_dirname/magick-(random).pdf` instead of `foo.pdf`. It seems that the backtick operator in the delegates configuration file is subtituted by an underscore (since https://github.com/ImageMagick/ImageMagick/commit/90ee513?). Example output for an unmodified configuration file:
```
$ convert -verbose foo.odt foo.pdf
"soffice" --headless --convert-to pdf --outdir `dirname "/tmp/magick-14078qGhBgaN3cBDC"` "/tmp/magick-14078qGhBgaN3cBDC" 2> "/tmp/magick-14078xpTW3hjb7USG"; mv "/tmp/magick-14078qGhBgaN3cBDC.pdf" "/tmp/magick-14078jKbgy9jt7znP"                                                                                                 
convert /tmp/magick-14078qGhBgaN3cBDC -> /tmp/pdf/_dirname/magick-14078qGhBgaN3cBDC.pdf using filter : writer_pdf_Export
mv: cannot stat ‘/tmp/magick-14078qGhBgaN3cBDC.pdf’: No such file or directory   
convert: delegate failed `"soffice" --headless --convert-to pdf --outdir `dirname "%i"` "%i" 2> "%Z"; mv "%i.pdf" "%o"' @ error/delegate.c/InvokeDelegate/1329.
convert: no decode delegate for this image format `ODT' @ error/constitute.c/ReadImage/535.    
convert: no images defined `x.pdf' @ error/convert.c/ConvertImageCommand/3212.
```

Anyway, even with when ``dirname &quot;%i&quot;`` is replaced by `/tmp`, conversion still fails (here an example for xlsx files):
```
$ convert -verbose foo.xlsx foo1.pdf
"soffice" --headless --convert-to pdf --outdir /tmp "/tmp/magick-4346u95gJrXCHVSq" 2> "/tmp/magick-4346PMDP9cxFPhTW"; mv "/tmp/magick-4346u95gJrXCHVSq.pdf" "/tmp/magick-4346lJNGQpJ840TY"
convert /tmp/magick-4346u95gJrXCHVSq -> /tmp/magick-4346u95gJrXCHVSq.pdf using filter : calc_pdf_Export
convert: no decode delegate for this image format `XLSX' @ error/constitute.c/ReadImage/535.
convert: no images defined `foo1.pdf' @ error/convert.c/ConvertImageCommand/3212.
```

The `%d` option which is documented at https://www.imagemagick.org/script/escape.php expands to an empty value so was of no use here.

This was tested with ImageMagick 6.9.1-2 Q16 x86_64 on Arch Linux.

**Comment by urban-warrior** [comment]
The backtick problem is fixed in ImageMagick 6.9.1-5 beta.   The delegate however is still failing and we are investigating.

**Comment by urban-warrior** [comment]
Ok, got a solution, available in ImageMagick 6.9.1-5 Beta later today.  Thanks.

**Comment by Lekensteyn** [comment]
For future reference, was this fixed in SVN rev 18768 (4bd9ce461ddb74edb081e7898950d4eba2d954ae)?

...oh apparently SVN rev 18742 (git commit dde5d36feffe4aea6b1cb10e723791ffff5a8967)?
